Dockerfile

```diff
@@ -0,0 +1,70 @@
+# Copyright 2023 Linode, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG TOOLCHAIN_VERSION=1.21
+
+#########################################################################################
+# Build
+#########################################################################################
+
+# First stage: building the driver executable.
+FROM docker.io/library/golang:${TOOLCHAIN_VERSION} as builder
+
+WORKDIR /work
+
+# Copy the Go Modules manifests.
+COPY go.mod go.mod
+COPY go.sum go.sum
+
+# Cache dep before building and copying source so that we don't need to re-download as
+# much and so that source changes don't invalidate our downloaded layer.
+RUN go mod download
+
+# Copy the go source.
+COPY cmd/ cmd/
+COPY internal/ internal/
+COPY pkg/ pkg/
+COPY Makefile Makefile
+
+# Explicitly set the version, so the make won't try to get it (and fail).
+ENV VERSION="builder"
+
+# Build.
+RUN make build
+
+#########################################################################################
+# Runtime
+#########################################################################################
+
+# Second stage: building final environment for running the executable.
+FROM gcr.io/distroless/static-debian11:latest AS runtime
+
```

Comment:
Is it possible to run the driver as a non root user? 

If so `USER 65532:65532` is good.

Reply:
Well, yes and no. I don't have much experience with the distroless images, and don't really know how to create `/var/lib/cosi` so the driver has enough permissions to create a socket there.

Reply:
The last commit should works as expected (container starts and binds a socket), with the final user being (as you mentioned) `USER 65532:65532`.